Cython/Compiler/Scanning.py

```diff
@@ -462,20 +469,21 @@ def next(self):
 
     def peek(self):
         saved = self.sy, self.systring
+        saved_pos = self.position()
         self.next()
         next = self.sy, self.systring
-        self.unread(*next)
+        next_pos = self.position()
+        self.unread(*(next+next_pos[1:]))
```

Comment:
This seems unnecessarily complex and unclear.
```suggestion
        self.unread(self.sy, self.systring, next_pos[1], next_pos[2])
```